lib/login_gov/hostdata/railtie.rb

```diff
@@ -0,0 +1,42 @@
+begin
+  require 'lograge'
+rescue LoadError => e
+  STDERR.puts 'lograge is required by the LoginGov::Hostdata::Railtie, please add it to the Gemfile'
+  raise e
+end
+
+if Gem::Version.new(Rails::VERSION::STRING) < Gem::Version.new('6.1')
+  raise 'LoginGov::Hostdata::Railtie needs rails >= 6.1 to log request info, please upgrade'
+end
+
+require 'login_gov/hostdata/log_formatter'
+require 'securerandom'
+
+module LoginGov
+  module Hostdata
+    class Railtie < Rails::Railtie
+      config.lograge.enabled = true
+
+      config.log_formatter = if Rails.env.development?
+        LoginGov::Hostdata::DevelopmentLogFormatter
+      else
+        LoginGov::Hostdata::LogFormatter
+      end
+
+      if Rails.env.development? || Rails.env.production?
+        config.lograge.formatter = Lograge::Formatters::Json.new
+      end
+
+      config.lograge.custom_options = lambda do |event|
+        event.payload[:timestamp] = Time.zone.now.iso8601
+        event.payload[:uuid] = SecureRandom.uuid
+        event.payload[:pid] = Process.pid
+        event.payload[:user_agent] = event.payload[:request].user_agent
+        event.payload[:ip] = event.payload[:request].remote_ip
+        event.payload[:host] = event.payload[:request].host
```

Comment:
anything that comes from `payload[:request]` is what needs Rails 6.1

Reply:
I tried to find some resources about what one can confidently expect to exist in `event.payload`, but... is it a certainty that `:request` and `:headers` will always be there?

Reply:
After some digging, I believe here the source of the payload in Rails:

https://github.com/rails/rails/blob/291a3d2ef29a3842d1156ada7526f4ee60dd2b59/actionpack/lib/action_controller/metal/instrumentation.rb#L19-L29

via

https://github.com/rails/rails/blob/4f1f63b98a866b573df5e26f05f167a6a9f4b797/activesupport/lib/active_support/notifications.rb